spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -78,16 +73,9 @@ private void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consum
 		}
 
 		try {
-			Message message;
-			if (this.messageConverter != null) {
-				message = this.messageConverter.toMessage(pubsubMessage.getData().toByteArray(),
-						new MessageHeaders(messageHeaders));
-			}
-			else {
-				message = MessageBuilder.withPayload(pubsubMessage.getData().toByteArray())
-						.copyHeaders(messageHeaders).build();
-			}
-			sendMessage(message);
+			sendMessage(
+					MessageBuilder.withPayload(pubsubMessage.getData().toByteArray())
```

Comment:
Well, and this way you do nothing, but just send a `byte[]` as a payload. In other words you don't do any conversions at all.
Is it what we would like to do here on the *inbound* side?

Reply:
Not quite. No conversions **on the adapter itself**, we just reuse `PubSubTemplate`'s conversion, instead of literally duplicating it. (Now you're not looking at my commit message ;) )

If we ever want to have conversion happening in the adapter, I think we should start using those `InboundMessageConverter`/`OutboundMessageConverter`. But, for now, I believe this is the better state.

Reply:
??? Show me, please, where is the conversion here `MessageBuilder.withPayload(pubsubMessage.getData().toByteArray())`.
You receive a `PubsubMessage` and get its `body` as a `byte[]`. 
What am I missing?

Your change doesn't reflect what you say in commit message. 🤷‍♂️ 

Sorry, if I'm so ignorant, but the change doesn't look like flexible and the goal we are trying to achieve with the conversion.

Reply:
This is embarrassing, but I thought I was commenting on `PubSubMessageHandler` :no_mouth: